langchain/utilities/wolfram_alpha.py

```diff
@@ -0,0 +1,59 @@
+"""Util that calls WolframAlpha."""
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.utils import get_from_dict_or_env
+
+
+class WolframAlphaAPIWrapper(BaseModel):
+    """Wrapper for Wolfram Alpha.
+
+    Docs for using:
+
+    1. Go to wolfram alpha and sign up for a developer account
+    2. Create an app and get your APP ID
+    3. Save your APP ID into WOLFRAM_ALPHA_APPID env variable
+    4. pip install wolframalpha
+
+    """
+
+    wolfram_client: Any  #: :meta private:
+    wolfram_alpha_appid: Optional[str] = None
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        wolfram_alpha_appid = get_from_dict_or_env(
+            values, "wolfram_alpha_appid", "WOLFRAM_ALPHA_APPID"
+        )
+        values["wolfram_alpha_appid"] = wolfram_alpha_appid
+
+        try:
+            import wolframalpha
+
+        except ImportError:
+            raise ImportError(
+                "wolframalpha is not installed. "
+                "Please install it with `pip install wolframalpha`"
+            )
+        client = wolframalpha.Client(wolfram_alpha_appid)
+
+        values["wolfram_client"] = client
+
+        # TODO: Add error handling if keys are missing
+        return values
+
+    def run(self, query: str) -> str:
+        """Run query through WolframAlpha and parse result."""
+        res = self.wolfram_client.query(query)
+        # Includes only text from the response
```

Comment:
@nickscamara can we return more from the Wolfram response (e.g. the units of the answer)? For example, if you run a query like "How big is the moon?" via LangChain's wrapper, you get back "1079.6", and then LangChain will return "The moon is 1079.6 miles in diameter.", which is incorrect - 1079.6 is actually the *radius*, not the diameter. 

I'm assuming the Wolfram response included that, but it was just dropped by the wrapper?

Reply:
Thank you for that! #584 should now include the assumption.